include/cuco/detail/static_map/static_map_ref.inl

```diff
@@ -594,6 +713,8 @@ class operator_impl<
     auto probing_iter      = probing_scheme(key, storage_ref.window_extent());
     auto const empty_value = ref_.impl_.empty_slot_sentinel().second;
 
+    auto constexpr use_insert = PayloadWrite and (sizeof(value_type) > 8);
```

Comment:
The `sizeof(value_type) > 8` just checks if we use `packed_cas` or not, right? This value might change based on the architecture (on sm_90 we have 16B CAS). It's fine for now but we have to find a better solution to the problem, e.g., based on a trait system or making it queryable from the `slot_inserter` (see [comment](https://github.com/NVIDIA/cuCollections/issues/554#issuecomment-2243481875)).